core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ExpressionRewriteRuleSet.java

```diff
@@ -304,9 +304,11 @@ public Result apply(ValuesNode valuesNode, Captures captures, Context context)
                 Expression rewritten;
                 if (row instanceof Row value) {
                     // preserve the structure of row
-                    rewritten = new Row(value.items().stream()
-                            .map(item -> rewriter.rewrite(item, context))
-                            .collect(toImmutableList()));
+                    ImmutableList.Builder<Expression> rowValues = ImmutableList.builderWithExpectedSize(value.items().size());
```

Comment:
original looks nicer.

Reply:
Yeah but resizes the list instead of preallocating it

Reply:
> Returns a new builder, expecting the specified number of elements to be added.
If expectedSize is exactly the number of elements added to the builder before ImmutableList.Builder.build is called, the builder is likely to perform better than an unsized builder() would have.

Reply:
Nicer != more performant :P

Reply:
Meh - I doubt it matters at all, so I would keep nicer. I will not fight though.